packages/app-spec-automatic-chess/src/libraries/squadChess.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * SQUAD CHESS RULES ENGINE
+ * 
+ * This library takes in a game state (board position, turn number, and legal moves) and an action (a move), 
+ * then returns a new state and set of legal moves.
+ * 
+ * General architecture: Take in a board state and an action (move), 
+ * return a new state and a list of legal moves.
+ * 
+ * DATA TYPES
+ * move = { from: [0,1], to: [4,6] }
+ * state = { position: {'0,1': {pieceId, player: 0||1}... }, turn: 14 }
+ * PIECES = { pieceId: { name: 'rook', mechanics: { 'move': [moveInputs] }}} (come from 'component' definitions)
+ * MECHANICS = { 'mechanic name': function that takes move inputs, returns moves } (come from 'game' definition)
+ * 
+ */
+
+const MECHANICS = {
+  'move': (inputs, from, position, turn) => {
```

Comment:
What about `params` or `mechanicParams`? Since this data is a list of different inputs that get used for the same mechanic. I think I prefer `params` to `config`, because `config` makes me think of something that stays the same once set, and here we have each of the inputs in turn being plugged into the mechanic. Don't have a strong opinion, though, this is a pretty specific data set (in a definition, it would be in `data: { mechanics: { mechanicName : [params] }}`).

Reply:
I slightly prefer config (though param is totally fine too) just because when writing the component data structure, the fact that `mechanicName` is a function is an implementation detail.

I guess it comes down to whether our mechanics configurable or parameterized?
and those are just two ways of saying basically the same thing.

Reply:
They do stay the same once set for each component